move/axelar_gateway/sources/gateway.move

```diff
@@ -164,6 +168,16 @@ public fun borrow_mut(
     table::borrow_mut(&mut self.messages, command_id)
 }
 
+// -----------
+// Macros
+// -----------
+macro fun peel_data<$T>($data: vector<u8>, $peel_fn: |&mut _| -> $T): $T {
+    let mut bcs = bcs::new($data);
+    let result = $peel_fn(&mut bcs);
```

Comment:
```suggestion
    let result = $T::peel(&mut bcs);
```
call peel on the type directly? If this is possible, I'd create two macro variants, one is `peel!` that always uses `T::peel` and another `peel_fn!` that uses the provided function. Might need to mention types explicitly to make `T::peel` work if it can't infer it

Reply:
probably not becayse $T is a type, but we need $module::peel.

Reply:
unfortunately, this is not possible